### ISSUE TITLE
"Checksum validation failed" when running Linkage Checker for spring-cloud-gcp

### ISSUE THREAD

**Issue by suztomo** [GoogleCloudPlatform/cloud-opensource-java#1417]
From https://github.com/spring-cloud/spring-cloud-gcp/pull/1658#issuecomment-631578103 . 

For https://github.com/spring-cloud/spring-cloud-gcp/pull/1658#issuecomment-631623638. The URLs are accessible from my browser.

```
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ spring-cloud-gcp-pubsub-stream-binder-functional-sample-test ---
[INFO] Installing /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-binder-functional-sample/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/target/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.3.BUILD-SNAPSHOT.jar to /home/travis/.m2/repository/org/springframework/cloud/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/1.2.3.BUILD-SNAPSHOT/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.3.BUILD-SNAPSHOT.jar
[INFO] Installing /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-binder-functional-sample/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/pom.xml to /home/travis/.m2/repository/org/springframework/cloud/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/1.2.3.BUILD-SNAPSHOT/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.3.BUILD-SNAPSHOT.pom
[WARNING] Could not validate integrity of download from http://www.terracotta.org/download/reflector/releases/org/slf4j/slf4j-api/maven-metadata.xml: Checksum validation failed, expected a19ee04115622963d6ca3efb5509550b9c097e5e but is 52fec9d8dfd13ef3f5e592604e6e3e6d7d19a9f3
[WARNING] Could not validate integrity of download from https://repo1.maven.org/maven2/org/slf4j/slf4j-jcl/1.0.1/slf4j-jcl-1.0.1.pom: Checksum validation failed, expected 7035ae7774a9a082a316a6943bbad9dfab6319b3 but is c5c0a3fff6071a4c720f1b7aa1b66cb9d0b26a21
[WARNING] Could not validate integrity of download from https://repo1.maven.org/maven2/org/opensaml/opensaml/maven-metadata.xml: Checksum validation failed, expected b50e385d27683179ca2de5043f96c1252301aca5 but is 7576ac94d82160710ee133d885e0ce71339e0ef1
[WARNING] Could not validate integrity of download from https://repo.maven.apache.org/maven2/org/opensaml/opensaml/maven-metadata.xml: Checksum validation failed, expected b50e385d27683179ca2de5043f96c1252301aca5 but is 7576ac94d82160710ee133d885e0ce71339e0ef1
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

When I tried to build spring-cloud-gcp from my Mac home internet. These are errors but it means the domain name is not resolvable.

```
[INFO] ------< org.springframework.cloud:spring-cloud-gcp-dependencies >-------
[INFO] Building Spring Cloud GCP Dependencies 1.2.3.BUILD-SNAPSHOT      [32/80]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ spring-cloud-gcp-dependencies ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-linkage-checker) @ spring-cloud-gcp-dependencies ---
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/org/jruby/jruby-core/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is b9478a389f97ad6c5e45f6fcf5883fb8467593a8
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/com/github/jnr/jnr-constants/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is ad3b997ba63ad8b9d1ed8767fdcb32849db9427a
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/org/jruby/joni/joni/2.1.1-SNAPSHOT/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 97ec7a31289cf2d4f158299577ba296db1df6621
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/org/jruby/joni/joni/2.1.1-SNAPSHOT/joni-2.1.1-SNAPSHOT.pom: Checksum validation failed, expected <!DOCTYPE but is c31e1a5cb08bbbcc5d424f86049fca87049abf04
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/org/jruby/jruby-core/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is b9478a389f97ad6c5e45f6fcf5883fb8467593a8
[WARNING] Could not validate integrity of download from http://www.terracotta.org/download/reflector/releases/org/slf4j/slf4j-api/maven-metadata.xml: Checksum validation failed, expected a19ee04115622963d6ca3efb5509550b9c097e5e but is 52fec9d8dfd13ef3f5e592604e6e3e6d7d19a9f3
[WARNING] Could not validate integrity of download from https://repo1.maven.org/maven2/org/opensaml/opensaml/maven-metadata.xml: Checksum validation failed, expected b50e385d27683179ca2de5043f96c1252301aca5 but is 7576ac94d82160710ee133d885e0ce71339e0ef1
[WARNING] Could not validate integrity of download from https://repo.maven.apache.org/maven2/org/opensaml/opensaml/maven-metadata.xml: Checksum validation failed, expected b50e385d27683179ca2de5043f96c1252301aca5 but is 7576ac94d82160710ee133d885e0ce71339e0ef1
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/com/github/jnr/jffi/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 003a584e642a2c3e32de796abe6b56a1e208716a
[WARNING] Could not validate integrity of download from http://repo2.maven.org/maven2/com/github/jnr/jnr-x86asm/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 265fec925bd8aac09becabf8788c810bbab8930b
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/commons-collections/commons-collections/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 9ff21001b309a49b8dd12e70a183eba1f5a2e454
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/commons-digester/commons-digester/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 410f8009407d2f866337457f0e06bb1ef9279c9e
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/commons-logging/commons-logging/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is c9bf2e54b36ad366f41407e25c2b84e2dc53a14e
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/org/hibernate/hibernate/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is c1bb5a78c4b2230a5b36612736ecd8f8a49a1195
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/hsqldb/hsqldb/1.8.0.2/hsqldb-1.8.0.2.pom: Checksum validation failed, expected <!DOCTYPE but is 78f7427520e463207543fb05f5898878ccd0af22
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/com/lowagie/itext/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 0697659b067e28e3823a990dad5db90560c5ecfa
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/jfree/jcommon/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is dbd0ef299305ba0283b77f958d47cfc219d57083
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/jfree/jfreechart/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 2e90d766d49185e615f48d497fb9d0c003d2d40f
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/xalan/xalan/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 4ce58e2ba7fcd1e514a473184a14e029334027ce
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/eclipse/jdtcore/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is 36278b46388799dbc0fae93d2027de94ce559957
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/ant/ant/maven-metadata.xml: Checksum validation failed, expected <!DOCTYPE but is cef0202106aef5358e57a149fcbfa86dcb5dbe28
[WARNING] Could not validate integrity of download from http://www.jasperforge.org/maven2/mondrian/mondrian/2.3.2.8944/mondrian-2.3.2.8944.pom: Checksum validation failed, expected <!DOCTYPE but is f8109b29af68bf8c8658d5fc2c08bd0a03536664

```

My ISP Verizon inserts HTML for unknown domains.

```
suztomo-macbookpro44% curl http://repo2.maven.org/maven2/com/github/jnr/jnr-x86asm/maven-metadata.xml
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><head><meta http-equiv="refresh" content="0;url=https://searchassist.verizon.com/main?ParticipantID=euekiz39ksg8nwp7iqj2fp5wzfwi5q76&FailedURI=http%3A%2F%2Frepo2.maven.org%2Fmaven2%2Fcom%2Fgithub%2Fjnr%2Fjnr-x86asm%2Fmaven-metadata.xml&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=us"/><script type="text/javascript">url="https://searchassist.verizon.com/main?ParticipantID=euekiz39ksg8nwp7iqj2fp5wzfwi5q76&FailedURI=http%3A%2F%2Frepo2.maven.org%2Fmaven2%2Fcom%2Fgithub%2Fjnr%2Fjnr-x86asm%2Fmaven-metadata.xml&FailureMode=1&Implementation=&AddInType=4&Version=pywr1.0&ClientLocation=us";if(top.location!=location){var w=window,d=document,e=d.documentElement,b=d.body,x=w.innerWidth||e.clientWidth||b.clientWidth,y=w.innerHeight||e.clientHeight||b.clientHeight;url+="&w="+x+"&h="+y;}window.location.replace(url);</script></head><body></body></html>%  
```

**Comment by elharo** [comment]
These look like errors when an artifact can't be downloaded, either permanently or temporarily. I don't think this is a bug in the linkage checker.

**Comment by suztomo** [comment]
I agree. The problem I wrote was for missing artifacts. I've updated the issue description with the errors from accessible URLs. https://github.com/spring-cloud/spring-cloud-gcp/pull/1658
(But the output is not in maven-install-plugin, not enforcer rule's)

**Comment by suztomo** [comment]
> [WARNING] Could not validate integrity of download from https://repo1.maven.org/maven2/org/slf4j/slf4j-jcl/1.0.1/slf4j-jcl-1.0.1.pom: Checksum validation failed, expected 7035ae7774a9a082a316a6943bbad9dfab6319b3 but is c5c0a3fff6071a4c720f1b7aa1b66cb9d0b26a21

https://repo1.maven.org/maven2/org/slf4j/slf4j-jcl/1.0.1/slf4j-jcl-1.0.1.pom.sha1 has 
```
7035ae7774a9a082a316a6943bbad9dfab6319b3  /home/projects/maven/repository-staging/to-ibiblio/maven2/org/slf4j/slf4j-jcl/1.0.1/slf4j-jcl-1.0.1.pom
```

`curl` and `sha1sum` says otherwise. It seems the warning message is valid.

```
suztomo@suztomo:~/spring-cloud-gcp$ curl https://repo1.maven.org/maven2/org/slf4j/slf4j-jcl/1.0.1/slf4j-jcl-1.0.1.pom |sha1sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   394  100   394    0     0   4061      0 --:--:-- --:--:-- --:--:--  4104
c5c0a3fff6071a4c720f1b7aa1b66cb9d0b26a21  -
```

**Comment by elharo** [comment]
OK, so what, if anything, should we do about this? It doesn't seem to be our error and is a true positive, though not of the sort we're looking for. Can we give any better error message in this case? The existing message seems fairly clear. Maybe this is WAI?

**Comment by suztomo** [comment]
I also think this is working as expected. Because Linkage Checker digs further in optional dependencies than normal Maven invocation, it found discrepancy in the hash, which otherwise unnoticed.

https://github.com/spring-cloud/spring-cloud-gcp/pull/1658#issuecomment-631578103

@meltsufin I created https://issues.sonatype.org/browse/MVNCENTRAL-5833 . Closing this warning message issue for now. Still I'm troubleshooting #1418 .
